test/e2e/upgrades/ingress.go

```diff
@@ -122,4 +153,98 @@ func (t *IngressUpgradeTest) verify(f *framework.Framework, done <-chan struct{}
 	}
 	By("hitting the Ingress IP " + t.ip)
 	framework.ExpectNoError(framework.PollURL(fmt.Sprintf("https://%v/", t.ip), "", framework.LoadBalancerPollTimeout, t.jig.PollInterval, t.httpClient, false))
+
+	// Note: This logic is dependant on the "static-ip" ingress manifest being
+	// used to create the ingress for this test. If the manifest is changed,
+	// this logic can/will break.
+	if triggerSync {
+		// We want to manually trigger a sync because then we can easily verify
+		// a correct sync completed after update.
+		By("updating ingress spec to manually trigger a sync")
+		t.jig.Update(func(ing *extensions.Ingress) {
+			hirv := extensions.HTTPIngressRuleValue{[]extensions.HTTPIngressPath{}}
+			hirv.Paths = append(
+				hirv.Paths,
+				extensions.HTTPIngressPath{
+					"/bar",
+					extensions.IngressBackend{"echoheadersx", intstr.FromInt(80)}})
+			ing.Spec.Rules = append(
+				ing.Spec.Rules,
+				extensions.IngressRule{
+					"foo.baz.com",
+					extensions.IngressRuleValue{&hirv}})
+
+		})
+		// Verifying our update to the ingress was synced to cloud.
+		wait.PollImmediate(framework.Poll, framework.SingleCallTimeout, func() (bool, error) {
+			umList := t.gceController.ListUrlMaps()
+			// The original spec (before update) had no host rules, so when we
+			// get host rules, we know the sync with cloud has occured.
+			if len(umList[0].HostRules) > 0 {
+				return true, nil
+			} else {
+				return false, nil
+			}
+		})
+		By("comparing GCP resources post-upgrade")
+		postUpgradeResourceStore := &GCPResourceStore{}
+		t.populateGCPResourceStore(postUpgradeResourceStore)
+		framework.ExpectNoError(compareGCPResourceStores(t.resourceStore, postUpgradeResourceStore, func(v1 reflect.Value, v2 reflect.Value) error {
+			i1 := v1.Interface()
+			i2 := v2.Interface()
+			// Special check for the UrlMap since we purposefully modified it.
+			if v1.Type() == reflect.TypeOf([]*compute.UrlMap{}) {
+				umList2 := i2.([]*compute.UrlMap)
+				if umList2[0].HostRules[0].Hosts[0] == "foo.baz.com" {
+					return nil
+				} else {
+					// TODO(rramkumar): Make this error message more descriptive.
+					return fmt.Errorf("forced update of UrlMap did not result in proper sync to cloud")
```

Comment:
What about return without the else block?
``` golang
if ... {
   return nil
}
return err
```

Reply:
Done